packages/datasource-replica/test/integrations/pull-delta.test.ts

```diff
@@ -0,0 +1,297 @@
+      expect(pullDeltaHandler).toHaveBeenCalledTimes(3);
+    });
+
+    describe('when a record is deleted', () => {
+      it('should execute the delta', async () => {
+        const pullDeltaHandler: ReplicaDataSourceOptions['pullDeltaHandler'] = jest
+          .fn()
+          .mockImplementationOnce(async () => {
+            return {
+              more: false,
+              newOrUpdatedEntries: [],
+              nextDeltaState: 'delta-state',
+              deletedEntries: [{ collection: 'contacts', record: { id: 1 } }],
+            } as PullDeltaResponse;
+          });
+
+        const datasource = await makeReplicaDataSource({
+          pullDeltaHandler,
+          schema: makeSchemaWithId('contacts'),
+          pullDeltaOnBeforeAccess: true,
+        });
+
+        expect(await getAllRecords(datasource, 'contacts')).toEqual([]);
+        expect(pullDeltaHandler).toHaveBeenCalledTimes(1);
+      });
+    });
+  });
+
+  describe('when the delta does not respect the schema', () => {
+    describe('when the collection name does not exist in the schema', () => {
+      it('should error log or an error', async () => {
+        const pullDeltaHandler: ReplicaDataSourceOptions['pullDeltaHandler'] = jest
+          .fn()
+          .mockImplementationOnce(() => {
+            return {
+              more: false,
+              newOrUpdatedEntries: [{ collection: 'notExist', record: { id: 1 } }],
+              nextDeltaState: 'delta-state',
+              deletedEntries: [],
+            } as PullDeltaResponse;
+          });
+
+        const logger = jest.fn();
+
+        const datasource = await makeReplicaDataSource(
+          {
+            pullDeltaHandler,
+            schema: makeSchemaWithId('contacts'),
+            pullDeltaOnBeforeAccess: true,
+          },
+          logger,
+        );
+
+        expect(await getAllRecords(datasource, 'contacts')).toEqual([]);
+
+        const errorEntries = logger.mock.calls.filter(entry => entry[0] === 'Error');
+
+        expect(errorEntries.toString()).toContain("Collection 'notExist' not found in schema");
+      });
+    });
+  });
+});
+
+describe('on schedule', () => {
+  it('should add scheduler on handler', async () => {
+    const schedulerStop = jest.fn();
+    Croner.Cron = jest.fn().mockImplementation((pattern, func) => {
+      func();
+
+      return { stop: schedulerStop };
+    }) as unknown as typeof Croner.Cron;
+
+    const pullDeltaHandler: ReplicaDataSourceOptions['pullDeltaHandler'] = jest
+      .fn()
+      .mockResolvedValueOnce({
+        more: false,
+        newOrUpdatedEntries: [{ collection: 'contacts', record: { id: 1 } }],
+        nextDeltaState: 'delta-state',
+        deletedEntries: [],
+      });
+
+    const datasource = await makeReplicaDataSource({
+      pullDeltaHandler,
+      schema: makeSchemaWithId('contacts'),
+      pullDeltaOnSchedule: '* * * * * *',
+      pullDeltaOnBeforeAccessDelay: 1,
+      pullDeltaOnBeforeAccess: true,
```

Comment:
Typically, this is an issue where `getAllRecords` triggers a pull delta, thus the test isn't testing the pullDeltaOnSchedule

Reply:
```suggestion
      pullDeltaOnSchedule: '* * * * * *',
```

and 

```
    expect(await getAllRecords(datasource, 'contacts')).toEqual([]);
    expect(await getAllRecords(datasource, 'contacts')).toEqual([{ id: 1 }]);
```

just like mister Moncel did on his test. Still relying on dark magic though, to force tick() the agent